src/commands/ProjectCommand.ts

```diff
@@ -0,0 +1,60 @@
+import Command from "../abstracts/Command";
+import {Message, MessageEmbed} from "discord.js";
+import {readFile} from "fs";
+
+export default class ProjectCommand extends Command {
+	private readonly fileDirectory = "./assets/projects.json";
+	constructor() {
+		super(
+			"project",
+			"Returns a random project idea based on given parameters.",
+		);
+	}
+
+	async run(message: Message, args: string[]): Promise<void> {
+		const embed = new MessageEmbed();
+		const query = args.map(arg => arg.toLowerCase()).filter(arg => arg.trim().length > 0);
+
+		if (args.length === 0) {
+			embed.setTitle("Projects")
+				.setColor("#add8e6")
+				.setDescription("Please provide arguments on the projects command.");
+			await message.channel.send(embed);
+		} else {
+			readFile(`${this.fileDirectory}`, {encoding: "utf8"}, async (error, projectData) => {
+				const projects: Array<Project> = JSON.parse(projectData);
+				const project = projects
+					.filter(this.removeTooLongDescriptions)
+					.filter(project => this.filterTags(project, query))
+					.sort(() => 0.5 - Math.random()).pop();
```

Comment:
I'm curious, what was your thought process being randomising the results?

Reply:
the resultset can be expansive, if a user would merely search on 'beginner' or 'easy', there could be 10 different projects applying. If there is no randomisation, they'll always retrieve the same project. By randomising, a lot more options are to be shown, and a lot more 'projects' can share the same tags.